### PR TITLE
feat: More fuzz tests

### PR DESCRIPTION
**Pull request by toninorair**
Copy pasting tests from separate repo

### CODE REVIEW

File: test/WrappedMToken.t.sol [hunk 1]
```diff
@@ -237,6 +347,122 @@ contract WrappedMTokenTests is Test {
...
+                    balance_ + accruedYield_,
+                    unwrapAmount_
+                )
+            );
+        }
+
+        vm.startPrank(_alice);
+        _wrappedMToken.unwrap(_alice, unwrapAmount_);
+
+        if ((unwrapAmount_ == 0) || (unwrapAmount_ > balance_)) return;
```
Comment (PierrickGT) [inline test/WrappedMToken.t.sol:405]: Should also include the accrued yield.
```suggestion
        if ((unwrapAmount_ == 0) || (unwrapAmount_ > balance_ + accruedYield_)) return;
```

File: test/WrappedMToken.t.sol [hunk 2]
```diff
@@ -261,6 +487,57 @@ contract WrappedMTokenTests is Test {
...
+        _mToken.setBalanceOf(address(_wrappedMToken), mBalance_);
+        _wrappedMToken.setTotalNonEarningSupply(totalNonEarningSupply_);
+        _wrappedMToken.setPrincipalOfTotalEarningSupply(principalOfTotalEarningSupply_);
+
+        _mToken.setCurrentIndex(index_);
+
+        uint240 excess = _wrappedMToken.claimExcess();
+
+        assertEq(_wrappedMToken.excess(), 0);
+        assertEq(_mToken.balanceOf(_wrappedMToken.vault()), excess); // TODO: Do with expectCall.
```
Comment (PierrickGT) [inline test/WrappedMToken.t.sol:538]: Not sure what this comment implies.
Should we test that `IMTokenLike(mToken).transfer(vault, excess_);` is called?

File: test/WrappedMToken.t.sol [hunk 3]
```diff
@@ -456,6 +733,109 @@ contract WrappedMTokenTests is Test {
...
+                    aliceBalance_ + aliceAccruedYield_,
+                    amount_
+                )
+            );
+        }
+
+        vm.prank(_alice);
+        _wrappedMToken.transfer(_bob, amount_);
+
+        if (amount_ > aliceBalance_) return;
```
Comment (PierrickGT) [inline test/WrappedMToken.t.sol:815]:
```suggestion
        if (amount_ > aliceBalance_ + aliceAccruedYield_) return;
```

**Review by PierrickGT — CHANGES_REQUESTED** [review]

File: test/WrappedMToken.t.sol (hunk 1 above)
Comment (toninorair) [reply test/WrappedMToken.t.sol:405]: this line is redundant by checks in 389-400

File: test/WrappedMToken.t.sol (hunk 2 above)
Comment (deluca-mike) [reply test/WrappedMToken.t.sol:538]: Yes. Not good practice to test state of Mocks.

File: test/WrappedMToken.t.sol (hunk 1 above)
Comment (PierrickGT) [reply test/WrappedMToken.t.sol]: On 389, it only checks that the event has been properly emitted. You still need to return early on line 405.

**Review by PierrickGT — APPROVED** [review]

**Pull request merged**